rpc/src/rpc.rs

```diff
@@ -2354,7 +2354,10 @@ fn encode_account<T: ReadableAccount>(
     data_slice: Option<UiDataSliceConfig>,
 ) -> Result<UiAccount> {
     if (encoding == UiAccountEncoding::Binary || encoding == UiAccountEncoding::Base58)
-        && account.data().len() > MAX_BASE58_BYTES
+        && data_slice
+            .map(|s| min(s.length, account.data().len() - s.offset))
```

Comment:
This will panic if a user supplies an offset that is larger than the length of any account in a set, which would be bad. Please use saturating math.

Reply:
Sorry, I don't know what that meme means.